### DISCUSSION TITLE
Hub and Spoke

### DISCUSSION THREAD

**Discussion by rmccar** [ONSdigital/design-system#714] [Feedback]
Use this place to discuss the [hub and spoke pattern](https://ons-design-system.netlify.com/patterns/hub-and-spoke/) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by matt-b-ur** [comment]
Does anyone know of an explanation of Hub and Spoke (the entry on the ONS Design System reports 'This pattern requires documentation.')?
I'm seeking **an explanation of what Hub and Spoke is in the context of questionnaire design**.
Matt.B (fingers crossed)

**Reply by jrbarnes9** [reply]
Hi @matt-b-ur .  It's on our backlog to add documentation on this pattern, but to answer your question now...
Hub and Spoke is the flow of answering sections of a questionnaire.  You answer the first section, then are shown the 'Hub' which is a list of sections to complete. You can choose any section next, and are sent off down a 'spoke' to answer the questions in that section.  Once complete, you return to the 'Hub' to choose the next section.  Once all 'spokes' are complete, you can view your answers via the 'Hub', or submit your questionnaire.

**Reply by jrbarnes9** [reply]
#1977

**Comment by jrbarnes9** [comment]
Check if we can remove the info panel from hub and spoke example. It was removed for census, and replaced with a warning panel only displayed once all sections are complete.

**Reply by jrbarnes9** [reply]
Raised #1754

**Comment by jrbarnes9** [comment]
Incorrect buttons in Hub and Spoke examples...
- 'Spoke - continue a section' example label should be 'Save and continue'
- All other 'continue' buttons should have a url and thus a chevron icon

**Comment by mr-nick17** [comment]
Could a mobile variant be created where the actions are anchored to the top right? Similar to the image ⬇️ 

<img width="382" alt="summary mobile view" src="https://user-images.githubusercontent.com/19624419/169529718-35a16eb0-d485-45d5-83a1-717d8f263f1f.png">

**Reply by jrbarnes9** [reply]
Raised #2274

**Comment by mr-nick17** [comment]
There's a (small) bug with focusing. When focusing on the action, the `:focus` fills the container (see image). Easily resolved with aligning the item `align-items: flex-start`. [PR](https://github.com/ONSdigital/dp-frontend-filter-flex-dataset/pull/71) for use on ONS website

<img width="724" alt="focus filling container" src="https://user-images.githubusercontent.com/19624419/194355126-ed941c29-375b-4325-b485-ddcec3e501c7.png">

**Reply by jrbarnes9** [reply]
Raised #2480
